private/js2ts/index.mjs

```diff
@@ -58,12 +58,13 @@ try {
 
 for await (const dirent of dir) {
   if (!dirent.isDirectory()) {
-    const { path: filepath } = dirent
-    const ext = extname(filepath)
+    const { name } = dirent
+    const ext = extname(name)
     if (ext !== '.js' && ext !== '.jsx') continue // eslint-disable-line no-continue
+    const filePath = resolve(dirent.path, name)
```

Comment:
This doesn't work on Node.js 18.x currently (because the value of `dirent.path` is different there). I've made a PR to node to deprecate that key and introduce a `parentPath` that would be stable across releases. In the mean time, we need to workaround that:
```suggestion
    const filePath = basename(dirent.path) === name ?
      dirent.path : // Some versions of Node.js give the full path as dirent.path.
      join(dirent.path, name) // Others supply only the path to the parent.
```